src/components/Mfa.tsx

```diff
@@ -0,0 +1,354 @@
+                        setBackupCodes(res.body.backupCodes)
+                    } else if (res.body.type === "Disabled") {
+                        setMfaStatus(res.body.type)
+                        setNewSecret(res.body.newSecret)
+                        setNewQr(res.body.newQr)
+                    }
+                } else {
+                    setMfaStatus("Error")
+                }
+            } catch (e) {
+                setMfaStatus("Error")
+                console.error(e)
+            }
+        }
+
+        fetchStatus()
+    }, [mfaStatus, mfaApi])
+
+    async function enableMfa() {
+        try {
+            setLoading(true)
+            const res = await mfaApi.mfaEnable({ code })
+            if (res.ok) {
+                setShowEnableModal(false)
+                setCode("")
+                setError(undefined)
+                setMfaStatus("Enabled")
+            } else {
+                res.error._visit({
+                    notFoundMfaEnable: () => setError(NOT_FOUND_MFA_ENABLE),
+                    badRequestMfaEnable: () => setError(BAD_REQUEST_MFA_ENABLE),
+                    forbiddenMfaEnable: () => setError(FORBIDDEN),
+                    unauthorized: () => setError(UNAUTHORIZED),
+                    _other: () => setError(UNEXPECTED_ERROR),
+                })
+            }
+        } catch (e) {
+            setError(UNEXPECTED_ERROR)
+            console.error(e)
+        } finally {
+            setLoading(false)
+        }
+    }
+
+    async function disableMfa() {
+        try {
+            setLoading(true)
+            const res = await mfaApi.mfaDisable()
+            if (res.ok) {
+                setShowDisableModal(false)
+                setError(undefined)
+                setMfaStatus("Disabled")
+            } else {
+                res.error._visit({
+                    notFoundMfaDisable: () => setError(NOT_FOUND_MFA_DISABLE),
+                    unauthorized: () => setError(UNAUTHORIZED),
+                    _other: () => setError(UNEXPECTED_ERROR),
+                })
+            }
+        } catch (e) {
+            setError(UNEXPECTED_ERROR)
+            console.error(e)
+        } finally {
+            setLoading(false)
+        }
+    }
+
+    function downloadBackupCodes() {
+        try {
+            const codesToText = backupCodes.join(" ")
+            const blob = new Blob([codesToText], { type: "text/plain" })
+            const url = URL.createObjectURL(blob)
+            const link = document.createElement("a")
+            link.href = url
+            link.download = "backup-codes.txt"
+            document.body.appendChild(link)
+            link.click()
+            document.body.removeChild(link)
+            window.URL.revokeObjectURL(url)
+        } catch (e) {
+            setError("Download failed")
+            console.error(e)
+        }
+    }
+
+    if (mfaStatus === "Loading" || mfaStatus === "Error") {
+        return null // ?
```

Comment:
Loading probably needs it's own state?

For error, we should have some generic "An unexpected error has occurred, please try again"

Reply:
[fix](https://github.com/PropelAuth/react/pull/11/commits#:~:text=handle%20loading%20%26%20error%20state%20for%20mfa)